src/base/ApplicationECSContainerDefinition.ts

```diff
@@ -92,81 +36,66 @@ export interface ApplicationECSContainerDefinitionProps {
 export function buildDefinitionJSON(
   config: ApplicationECSContainerDefinitionProps
 ): string {
-  let templateInstance = JSON_TEMPLATE;
-
-  // env vars default is [], whereas secrets default is null. makes sense.
-  const envVarsValue = config.envVars ? JSON.stringify(config.envVars) : '[]';
-  const secretEnvVarsValue = config.secretEnvVars
-    ? JSON.stringify(config.secretEnvVars)
-    : 'null';
+  const containerDefinition = {
+    dnsSearchDomains: null,
+    environmentFiles: null,
+    logConfiguration: {
+      logDriver: 'awslogs',
+      secretOptions: [],
+      options: {
+        'awslogs-group': config.logGroup,
+        'awslogs-region': 'us-east-1',
+        'awslogs-stream-prefix': 'ecs',
+      },
+    },
+    entryPoint: null,
+    portMappings: [
+      {
+        hostPort: config.hostPort,
+        protocol: config.protocol ?? 'tcp',
+        containerPort: config.containerPort,
+      },
+    ],
+    linuxParameters: null,
+    cpu: config.cpu ?? 0,
+    environment: config.envVars ?? [],
+    resourceRequirements: null,
+    ulimits: null,
+    repositoryCredentials: config.repositoryCredentialsParam
+      ? { credentialsParameter: config.repositoryCredentialsParam }
+      : null,
+    dnsServers: null,
+    mountPoints: [],
+    workingDirectory: null,
+    secrets: config.secretEnvVars ?? null, // env vars default is [], whereas secrets default is null. makes sense.
+    dockerSecurityOptions: null,
+    memory: null,
+    memoryReservation: config.memoryReservation ?? null,
+    volumesFrom: [],
+    stopTimeout: null,
+    image: config.containerImage,
+    startTimeout: null,
+    firelensConfiguration: null,
+    dependsOn: null,
+    disableNetworking: null,
+    interactive: null,
+    healthCheck: config.healthCheck ?? null,
+    essential: true,
+    links: null,
+    hostname: null,
+    extraHosts: null,
+    pseudoTerminal: null,
+    user: null,
+    readonlyRootFilesystem: false,
+    dockerLabels: null,
+    systemControls: null,
+    privileged: null,
+    name: config.name,
+  };
 
   if (config.command) {
-    templateInstance = templateInstance.replace(
-      '[[COMMAND]]',
-      '[' + config.command.map((c) => `"${c}"`).join(',') + ']'
-    );
-  } else {
-    templateInstance = templateInstance.replace('"command": [[COMMAND]],', '');
-  }
-
-  templateInstance = templateInstance.replace('[[LOG_GROUP]]', config.logGroup);
```

Comment:
Question: `logGroup` is optional, but this code will fail if it's not set, right? Should it be made required?